pyprotista/parsers/ident/comet_2020_01_4_parser.py

```diff
@@ -1,65 +1,135 @@
+                mapping_dict,
+            )
+
+        elif entry_tag == (f"{element_tag_prefix}DBSequence"):
+            stage = 1
+            # Short before peptide_lookup begins
+        elif entry_tag == (f"{element_tag_prefix}AdditionalSearchParams"):
+            stage = 2
+        elif entry_tag == (f"{element_tag_prefix}Inputs"):
+            stage = 3
+            # Short before spec_record begins
+
+        elif entry_tag == (f"{element_tag_prefix}AnalysisSoftware"):
+            version = "comet_" + "_".join(
+                re.findall(r"([/d]*\d+)", entry.attrib["version"])
+            )
+
+        entry.clear()
+        if entry_tag == (f"{element_tag_prefix}SpectrumIdentificationList"):
+            return (
+                version,
+                peptide_lookup,
+                spec_records,
+                modification_mass_map,
+                fixed_mods,
+            )
 
-        spec_records.append(psm_level_dict)
-    return pd.DataFrame(spec_records)
+
+def _peptide_lookup(entry, entry_tag, sequence, modifications, peptide_lookup):
+    if entry_tag == (f"{element_tag_prefix}PeptideSequence"):
+        sequence = {"sequence": entry.text}
+    elif entry_tag == (f"{element_tag_prefix}Modification"):
+        mod = {"monoisotopicMassDelta": entry.attrib["monoisotopicMassDelta"]}
+        mod.update({"location": entry.attrib["location"]})
+        modifications.append(mod)
+    elif entry_tag == (f"{element_tag_prefix}Peptide"):
+        if len(modifications) > 0:
+            peptide_lookup[entry.attrib["id"]] = {"modifications": modifications}
+        else:
+            peptide_lookup[entry.attrib["id"]] = {"modifications": ""}
+        peptide_lookup[entry.attrib["id"]].update(sequence)
+        modifications = []
+        sequence = ""
+    return sequence, modifications, peptide_lookup
+
+
+def _get_modifications(entry, entry_tag, modification_mass_map, mod_name, fixed_mods):
+    if entry_tag == (f"{element_tag_prefix}cvParam"):
+        mod_name = entry.attrib["name"]
+    elif entry_tag == (f"{element_tag_prefix}SearchModification"):
+        modification_mass_map[entry.attrib["massDelta"]] = mod_name
+        if entry.attrib["fixedMod"] == "true":
+            fixed_mods.update({entry.attrib["residues"]: mod_name})
+    return modification_mass_map, mod_name, fixed_mods
+
+
+def _spec_records(
+    entry, entry_tag, spec_results, spec_ident_items, spec_records, mapping_dict
+):
+    if entry_tag == (f"{element_tag_prefix}cvParam"):
+        if entry.attrib["name"] in mapping_dict:
+            spec_results.update(
+                {mapping_dict[entry.attrib["name"]]: entry.attrib["value"]}
+            )
+    elif entry_tag == (f"{element_tag_prefix}SpectrumIdentificationItem"):
+        for i in list(entry.attrib):
+            if i in mapping_dict:
+                spec_results.update({mapping_dict[i]: entry.attrib[i]})
+        spec_ident_items.append(spec_results)
+        spec_results = {}
```

Comment:
you update in the loop above just to reset it here? hmmm ... not sure I get it

Reply:
Well yes the thing is, there can be multiple SpectrumIdentificationItem - so i need to reset spec_results for the next one.
Also there are some general cvParams under SpectrumIdentificationResult, which belong to all SpectrumIdentificationItems (see for loop line 184). I reused the variable spec_results for these cvParam aswell.